src/views/SignUpView.vue

```diff
@@ -1,36 +1,79 @@
 <template>
   <main>
+    <error-toast v-show="isToastShown" text="Ooops, something went wrong. Try again." />
+
     <div class="py-12">
       <h2 class="mb-8 text-3xl text-center font-bold tracking-wider">Sign up</h2>
       <the-form>
         <template #inputs>
           <div class="flex flex-col px-6">
-            <the-input type="text" placeholder="User name"></the-input>
-            <the-input type="password" placeholder="Password"></the-input>
-            <the-input type="password" placeholder="Repeat password"></the-input>
-            <the-input type="email" placeholder="Email"></the-input>
+            <the-input
+            v-model="userData.name"
+            type="text"
+            placeholder="User name"
+            :conditions="isFormSubmitted && errorMessages.nameError !== ''"
+            :errorText="errorMessages.nameError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.passOne"
+            type="password"
+            placeholder="Password"
+            :conditions="isFormSubmitted && errorMessages.passOneError !== ''"
+            :errorText="errorMessages.passOneError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.passTwo"
+            type="password"
+            placeholder="Repeat password"
+            :conditions="isFormSubmitted && errorMessages.passTwoError !== ''"
+            :errorText="errorMessages.passTwoError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.email"
+            type="email"
+            placeholder="Email"
+            :conditions="isFormSubmitted && errorMessages.emailError !== ''"
+            :errorText="errorMessages.emailError"
+            >
+            </the-input>
           </div>
         </template>
         
         <template #buttons>
           <div class="flex justify-center">
-            <the-button text="Sign up" />
+            <the-button @click.prevent="signUpUser" text="Sign up" />
           </div>
         </template>
         
         <template #others>
-          <div class="flex justify-evenly px-6 mt-4 text-center">
-            <the-input type="checkbox" />
-            <label v-if="$route.name === 'sign-up'" class="text-sm">
+          <div class="flex justify-evenly px-6 mt-8 text-center">
+            <the-input
+            @input.capture="isCheckboxChecked = !isCheckboxChecked"
+            type="checkbox"
+            idValue="privacy"
+            />
+            <label v-if="$route.name === 'sign-up'" for="privacy" class="text-sm">
                 By clicking here, I state that I have read and understood the
                 <span class="border-b-brown-color border-b-2 text-brown-color font-bold">
-                  <router-link :to="{ name: 'terms-and-conditions' }" target="_blank">
+                  <router-link :to="{ name: 'privacy-policy' }" target="_blank">
```

Comment:
why do you set target _blank?

Reply:
Because if I were a user that wants to read the Privacy Policy before clicking the "Sign up" button but **after** filling out the form, I'd like this page to open in a separate tab. In another case, I would loose all my filling data and I would have to do it again.

Reply:
Then save the form data in localStorage or open the privacy policy inside a modal. Creating separate context is bad in terms of UX.

Reply:
You don't have to do it now. That could be quite a change so consider implementing it sooner than later

Reply:
**Added a modal**

However, could you explain why creating a separate page for Privacy Policy is bad  for UX? I seem not to get it...

Reply:
Creating a separate page for privacy policy is not bad at all! I'd even highly suggest to keep it. It's bad to create separate context (tab) for the user (in order to read it). This particular action should trigger a modal instead **if** user should be able to read it without making any step outside the form they're supposed to fill. Does it make sense now?

Reply:
So, basically, what you're saying is that I should keep the Privacy Policy as a page, in case someone types it into the address bar and/or because this should actually be a page but if I don't want a user to leave this sign-up page in order to read it, I should have this modal in addition to that?

Reply:
Correct. Making a task for it could be a good idea, if you plan to do it.

Reply:
**Added the Privacy Policy page again**

I'm sorry but I have no idea what do you mean by "making a task" 😥